File: apps/api/internal/dto/clinical_timeline.go
package dto

import "time"

// ClinicalTimelineSource identifica a origem de uma entrada do histórico de um item clínico.
const (
	ClinicalSourceEscoreLight = "escore_light" // Escore Plenya Light (sessão anônima, claimed)
	ClinicalSourcePreConsulta = "pre_consulta" // Formulário de preparação pré-consulta
	ClinicalSourceAnamnese    = "anamnese"     // Anamnese (itens registrados pela equipe na consulta)
)

// ClinicalTimelineEntry é um ponto na linha do tempo de um ScoreItem para um paciente,
// independente da origem (Escore Light, pré-consulta ou anamnese/consulta).
//
// @Description Entrada do histórico longitudinal de um item clínico
type ClinicalTimelineEntry struct {
	// Data do registro (consultation_date da anamnese, submit do prep, criação da sessão light)
	Date time.Time `json:"date"`

	// Origem: escore_light | pre_consulta | anamnese
	Source string `json:"source"`

	// Rótulo legível da origem (ex.: "Escore Light", "Pré-consulta", "Anamnese")
	SourceLabel string `json:"sourceLabel"`

	// Nível classificado (0-6), quando houver
	SelectedLevel *int `json:"selectedLevel,omitempty"`

	// Valor numérico medido, quando houver
	NumericValue *float64 `json:"numericValue,omitempty"`

	// Valor textual, quando houver
	TextValue *string `json:"textValue,omitempty"`
}

// ClinicalItemHistoryResponse é a resposta do histórico de um item clínico.
//
// @Description Histórico longitudinal de um item clínico (todas as origens)
type ClinicalItemHistoryResponse struct {
	ScoreItemID string                  `json:"scoreItemId"`
	Entries     []ClinicalTimelineEntry `json:"entries"`
}

File: apps/web/lib/api/clinical-timeline.ts
import { useQuery } from '@tanstack/react-query'
import { apiClient } from '../api-client'

// Origem de uma entrada do histórico clínico.
export type ClinicalTimelineSource = 'escore_light' | 'pre_consulta' | 'anamnese'

export interface ClinicalTimelineEntry {
  date: string
  source: ClinicalTimelineSource
  sourceLabel: string
  selectedLevel?: number
  numericValue?: number
  textValue?: string
  scaleResponses?: { answers?: Record<string, number>; total?: number; words?: string[] }
}

export interface ClinicalItemHistoryResponse {
  scoreItemId: string
  entries: ClinicalTimelineEntry[]
}

export function getItemHistory(patientId: string, scoreItemId: string) {
  return apiClient.get<ClinicalItemHistoryResponse>(
    `/api/v1/patients/${patientId}/score-items/${scoreItemId}/history`
  )
}

// Histórico longitudinal de um item clínico para um paciente (Escore Light + pré-consulta +
// anamnese). Lazy: só dispara quando habilitado (item expandido) e com paciente conhecido.
export function useItemHistory(
  patientId: string | null | undefined,
  scoreItemId: string,
  enabled: boolean
) {
  return useQuery({
    queryKey: ['clinical-timeline', patientId, scoreItemId],
    queryFn: () => getItemHistory(patientId!, scoreItemId),
    enabled: !!patientId && !!scoreItemId && enabled,
    staleTime: 60_000,
  })
}
